File: tutorial4-start/src-ios/Libraries/FoyerKit/Sources/FoyerServices/MockAccount.swift
// Copyright (c) 2026 Modaal.dev
// Licensed under the MIT License. See LICENSE file for details.

import FoyerKit
import Foundation

/// The account port as a mock service: the saved name is held in memory.
public final class MockAccount: NSObject, AccountPort {
  public private(set) var displayName: String?

  public override init() {}

  public func saveDisplayName(name: String, onSaved: @escaping () -> Void) {
    displayName = name
    onSaved()
  }
}

File: tutorial3-complete/src-kmp/app/src/main/kotlin/dev/modaal/foyer/app/services/MockAccount.kt
// Copyright (c) 2026 Modaal.dev
// Licensed under the MIT License. See LICENSE file for details.

package dev.modaal.foyer.app.services

import dev.modaal.foyer.ports.AccountPort

/** The account port as a mock service: the saved name is held in memory. */
class MockAccount : AccountPort {
  var displayName: String? = null
    private set

  override fun saveDisplayName(name: String, onSaved: () -> Unit) {
    displayName = name
    onSaved()
  }
}
